common/rpctypes/types.go

```diff
@@ -80,36 +86,47 @@ type MakeOfferRequest struct {
 
 // MakeOfferResponse ...
 type MakeOfferResponse struct {
-	ID string `json:"offerID"`
+	PeerID  peer.ID    `json:"peerID"`
+	OfferID types.Hash `json:"offerID"`
 }
 
 // SignerRequest initiates the signer_subscribe handler from the front-end
 type SignerRequest struct {
-	OfferID    string `json:"offerID"`
-	EthAddress string `json:"ethAddress"`
-	XMRAddress string `json:"xmrAddress"`
+	OfferID    types.Hash `json:"offerID"`
+	EthAddress string     `json:"ethAddress"`
+	XMRAddress string     `json:"xmrAddress"`
 }
 
 // SignerResponse sends a tx to be signed to the front-end
 type SignerResponse struct {
-	OfferID string `json:"offerID"`
-	To      string `json:"to"`
-	Data    string `json:"data"`
-	Value   string `json:"value"`
+	OfferID types.Hash `json:"offerID"`
+	To      string     `json:"to"`
+	Data    string     `json:"data"`
+	Value   string     `json:"value"`
 }
 
 // SignerTxSigned is a response from the front-end saying the given tx has been submitted successfully
 type SignerTxSigned struct {
-	OfferID string `json:"offerID"`
-	TxHash  string `json:"txHash"`
+	OfferID types.Hash     `json:"offerID"`
+	TxHash  ethcommon.Hash `json:"txHash"`
 }
 
 // BalancesResponse holds the response for the combined Monero and Ethereum Balances request
 type BalancesResponse struct {
-	MoneroAddress           string   `json:"monero_address"`
-	PiconeroBalance         uint64   `json:"piconero_balance"`
-	PiconeroUnlockedBalance uint64   `json:"piconero_unlocked_balance"`
-	BlocksToUnlock          uint64   `json:"blocks_to_unlock"`
-	EthAddress              string   `json:"eth_address"`
-	WeiBalance              *big.Int `json:"wei_balance"`
+	MoneroAddress           string   `json:"moneroAddress"`
+	PiconeroBalance         uint64   `json:"piconeroBalance"`
+	PiconeroUnlockedBalance uint64   `json:"piconeroUnlockedBalance"`
+	BlocksToUnlock          uint64   `json:"blocksToUnlock"`
+	EthAddress              string   `json:"ethAddress"`
+	WeiBalance              *big.Int `json:"weiBalance"`
+}
+
+// AddressesResponse ...
+type AddressesResponse struct {
+	Addrs []string `json:"addresses"`
+}
```

Comment:
This was moved, unchanged, from `rpc/net.go` so we have all the RPC types in one place.